### PR TITLE
Allow setting of maxBuffer for packing external modules

### PR DESCRIPTION
**Pull request by mhodgson**
I received the following error when trying to compile in Docker: `stderr maxBuffer exceeded`. It stems from the npm-programatic module and its use of `exec` when we install external dependencies. I have a PR opened there to add a setting for maxBuffer: https://github.com/Manak/npm-programmatic/pull/8

I don't recommend merging this until that PR is accepted, but I wanted to make this PR in the meantime for anyone else that runs into this problem.

This PR adds a setting called `packExternalModulesMaxBuffer` that is passed to the npm `exec` for installing external dependencies.

### CODE REVIEW

**Comment by mhodgson** [comment]
OK, the PR on npm-programmatic was accepted so this now uses a non-forked version. Ready to merge.

**Comment by gorjuspixels** [comment]
Running into a similar issue. I'm currently using @mhodgson's fork that works perfectly. Hey @thenikso, any plans on merging this soon?

**Comment by jelder** [comment]
@thenikso The other PR (https://github.com/Manak/npm-programmatic/pull/8) has been merged. Think we could get this one merged soon, too?

File: package.json [hunk 1]
```diff
@@ -31,7 +31,8 @@
     "body-parser": "^1.15.2",
     "express": "^4.14.0",
     "fs-extra": "^0.26.7",
-    "npm-programmatic": "0.0.5"
+    "npm-programmatic": "^0.0.7",
+    "webpack": "^1.13.1"
```
Comment (hassankhan) [inline package.json]: We shouldn't hardcode a Webpack dependency since #83 means a consumer has to install Webpack as a peer dependency.

**Comment by HyperBrain** [comment]
@mhodgson Can you update (rebase this PR) onto the current master? I think the issue still is present on Docker as nothing changed and the npm-programmatic version in master is still 0.0.5?

**Comment by mhodgson** [comment]
@HyperBrain all set.

**Comment by HyperBrain** [comment]
@mhodgson Great 👍 ! I will check it out tomorrow and do some tests (with master and the 3.0.0 branch).
If everything works well, I'll merge it into master.

**Comment by HyperBrain** [comment]
Tests against **master**, **v3.0.0** and **v3.0.0-individual-packaging** were successful. I also noticed a speed-up while packaging.

The PR is ready to be merged -> nevertheless I will wait. Depending on the status (quality/testing) of the individual packaging branch the merge target might be the v3.0.0 branch. If everything goes well, that will be released next week.

**Comment by HyperBrain** [comment]
@mhodgson If you check the "enable edits by maintainers" checkbox in the PR, I can add some README changes to the PR and retarget it before I'll merge it next week.

**Comment by mhodgson** [comment]
@HyperBrain checked.

**Comment by HyperBrain** [comment]
@mhodgson All v3 features are now merged into the `v3.0.0` branch. Unfortunately the merge left this PR with conflicts. IMO the easiest way would be to create a new PR with these changes (it's just 2 lines of code) from the latest upstream v3.0.0 branch. Can you do that, so that I can get it into v3?

**Comment by mhodgson** [comment]
@HyperBrain done: https://github.com/elastic-coders/serverless-webpack/pull/185

**Pull request closed without merging**